### PR TITLE
[SHELL32] Implement Move To Folder

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-11132](https://jira.reactos.org/browse/CORE-11132)

- Add context menu item `"Mo&ve to folder..."` and implement the action.
- Implement the `"Mo&ve to folder..."` menu item of `"Edit"` menu of Explorer.

### CODE REVIEW

File: dll/win32/shell32/lang/da-DK.rc [hunk 1]
```diff
@@ -1008,4 +1008,8 @@ BEGIN
     IDS_COPYTOTITLE "Select the place where you want to copy '%s'. Then click the Copy button."
     IDS_COPYITEMS "Copy Items"
     IDS_COPYBUTTON "Copy"
+    IDS_MOVETOMENU "Move to &folder..."
+    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Copy button."
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/da-DK.rc]: IDS_MOVETOTITLE "Wählen Sie den Ort, in den '%s' verschoben werden soll. Klicken Sie anschließend auf Verschieben."

File: dll/win32/shell32/lang/de-DE.rc [hunk 2]
```diff
@@ -1003,4 +1003,8 @@ BEGIN
     IDS_COPYTOTITLE "Wählen Sie den Ort, in den '%s' kopiert werden soll. Klicken Sie anschließend auf Kopieren."
     IDS_COPYITEMS "Elemente kopieren"
     IDS_COPYBUTTON "Kopieren"
+    IDS_MOVETOMENU "Move to &folder..."
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/de-DE.rc:1006]: IDS_MOVETOMENU "In Ordner &verschieben..."

File: dll/win32/shell32/lang/de-DE.rc [hunk 3]
```diff
@@ -1003,4 +1003,8 @@ BEGIN
...
+    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Move button."
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/de-DE.rc:1007]: IDS_MOVETOTITLE "Wählen Sie den Ort, in den '%s' verschoben werden soll. Klicken Sie anschließend auf Verschieben."

File: dll/win32/shell32/lang/de-DE.rc [hunk 4]
```diff
@@ -1003,4 +1003,8 @@ BEGIN
...
+    IDS_MOVEITEMS "Move Items"
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/de-DE.rc:1008]: IDS_MOVEITEMS "Elemente verschieben"

File: dll/win32/shell32/lang/de-DE.rc [hunk 5]
```diff
@@ -1003,4 +1003,8 @@ BEGIN
...
+    IDS_MOVEBUTTON "Move"
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/de-DE.rc:1009]: IDS_MOVEBUTTON "Verschieben"

File: dll/win32/shell32/lang/bg-BG.rc [hunk 6]
```diff
@@ -1003,4 +1003,8 @@ BEGIN
     IDS_COPYTOTITLE "Select the place where you want to copy '%s'. Then click the Copy button."
     IDS_COPYITEMS "Copy Items"
     IDS_COPYBUTTON "Copy"
+    IDS_MOVETOMENU "Move to &folder..."
+    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Copy button."
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/bg-BG.rc]: This line is wrong in all rc language files except the german one IDS_MOVETOTITLE "Then click the Copy button." should be "Then click the Move button." for all languages!

Comment (JoachimHenze) [reply dll/win32/shell32/lang/bg-BG.rc]: Ah good. You fixed that already! Nice!

Comment (katahiromz) [reply dll/win32/shell32/lang/bg-BG.rc]: OK.

File: dll/win32/shell32/lang/de-DE.rc (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/lang/de-DE.rc]: OK.

File: dll/win32/shell32/lang/de-DE.rc (hunk 3 above)
Comment (katahiromz) [reply dll/win32/shell32/lang/de-DE.rc]: OK.

File: dll/win32/shell32/lang/de-DE.rc (hunk 4 above)
Comment (katahiromz) [reply dll/win32/shell32/lang/de-DE.rc]: OK.

File: dll/win32/shell32/lang/de-DE.rc (hunk 5 above)
Comment (katahiromz) [reply dll/win32/shell32/lang/de-DE.rc]: OK.

File: dll/win32/shell32/lang/da-DK.rc [hunk 7]
```diff
@@ -1009,7 +1009,7 @@ BEGIN
     IDS_COPYITEMS "Copy Items"
     IDS_COPYBUTTON "Copy"
     IDS_MOVETOMENU "Move to &folder..."
-    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Move button."
+    IDS_MOVETOTITLE "Wählen Sie den Ort, in den '%s' verschoben werden soll. Klicken Sie anschließend auf Verschieben."
```
Comment (JoachimHenze) [inline dll/win32/shell32/lang/da-DK.rc:1012]: Sorry about this one. This is still not correct. My fault. I did not intend to propose a change to da-DK. It was an off-by-one from my side. Better just leave the english text here for now.

Comment (katahiromz) [reply dll/win32/shell32/lang/da-DK.rc]: OK.

**Comment by JoachimHenze** [comment]
Just a note for the sake of completeness (I will not rate this as a blocker from my side). When selecting multiple files on the desktop, the new "Copy to folder/Move to folder" should not be part of the context menu. It is not in XPSP3. That new feature is just available from the menubar of explorer filebrowser. That's different to the normal CopyTo/MoveTo which is available from context menu.

File: dll/win32/shell32/lang/ru-RU.rc [hunk 8]
```diff
@@ -1009,4 +1009,8 @@ BEGIN
     IDS_COPYTOTITLE "Select the place where you want to copy '%s'. Then click the Copy button."
     IDS_COPYITEMS "Copy Items"
     IDS_COPYBUTTON "Copy"
+    IDS_MOVETOMENU "Move to &folder..."
+    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Move button."
+    IDS_MOVEITEMS "Move Items"
+    IDS_MOVEBUTTON "Move"
```
Comment (JoseSanchezz) [inline dll/win32/shell32/lang/ru-RU.rc:1015]:
```
    IDS_MOVETOMENU "Переместить в папку..."
    IDS_MOVETOTITLE "Выберите место, куда вы хотите переместить '%s'. Затем нажмите кнопку 'Переместить.'"
    IDS_MOVEITEMS "Перемещение предметов"
    IDS_MOVEBUTTON "Переместить"
```

Comment (katahiromz) [reply dll/win32/shell32/lang/ru-RU.rc]: OK.

**Pull request merged**